src/settings/accounts-settings.test.tsx

```diff
@@ -0,0 +1,756 @@
+		const confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona1Row).not.toBeInTheDocument();
+	});
+
+	test('Should create only identities which have not been removed from the unsaved changes', async () => {
+		const defaultFirstName = faker.person.firstName();
+		const defaultLastName = faker.person.lastName();
+		const defaultFullName = faker.person.fullName({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultEmail = faker.internet.email({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultId = faker.string.uuid();
+
+		const identitiesArray: Account['identities']['identity'] = [
+			{
+				id: defaultId,
+				name: 'DEFAULT',
+				_attrs: {
+					zimbraPrefIdentityName: defaultFullName,
+					zimbraPrefReplyToEnabled: 'FALSE',
+					zimbraPrefFromDisplay: defaultFirstName,
+					zimbraPrefFromAddress: defaultEmail,
+					zimbraPrefIdentityId: defaultId
+				}
+			}
+		];
+
+		const account: Account = {
+			name: defaultEmail,
+			rights: { targets: [] },
+			signatures: { signature: [] },
+			id: defaultId,
+			displayName: '',
+			identities: {
+				identity: identitiesArray
+			}
+		};
+
+		const identitiesDefault = map(identitiesArray, (item, index) =>
+			identityToIdentityProps(item, index)
+		);
+
+		identitiesDefault.unshift(identitiesDefault.pop() as IdentityProps);
+
+		server.use(
+			rest.post('/service/soap/BatchRequest', (req, res, ctx) =>
+				res(
+					ctx.json({
+						Body: {
+							BatchResponse: {
+								CreateIdentityResponse: []
+							}
+						}
+					})
+				)
+			)
+		);
+
+		const pendingBatchRequest = waitForRequest('POST', '/service/soap/BatchRequest');
+
+		const { user } = setup(
+			<AccountsSettings account={account} identitiesDefault={identitiesDefault} />
+		);
+		await screen.findByText('sendAs');
+
+		await user.click(screen.getByRole('button', { name: /add persona/i }));
+		const persona1Row = screen.getByText('New Persona 1');
+		expect(persona1Row).toBeVisible();
+
+		await user.click(screen.getByRole('button', { name: /add persona/i }));
+		const persona2Row = screen.getByText('New Persona 2');
+		expect(persona2Row).toBeVisible();
+
+		await user.click(screen.getByRole('button', { name: /add persona/i }));
+		const persona3Row = screen.getByText('New Persona 3');
+		expect(persona3Row).toBeVisible();
+
+		await user.click(persona1Row);
+		await user.click(screen.getByRole('button', { name: /delete/i }));
+		let confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona1Row).not.toBeInTheDocument();
+
+		await user.click(persona2Row);
+		await user.click(screen.getByRole('button', { name: /delete/i }));
+		confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona2Row).not.toBeInTheDocument();
```

Comment:
:warning: This test is not behaving as you might think. Try replacing line 389 to run the expect with the string value instead of the element (screen.getByText('New Persona 1')) and you wiil see the test fails. This is because persona1Row is no more in the document after the creation of the second identity! Try by putting an expect(persona1Row).toBeInTheDocument() right after line 388.

So basically you are not clicking on anything on lines 394 and 400, and you are deleting persona 3 first and persona 2 after that, because of the automatic selection of the previous item. And the expect of line 398 and 404 are successful because the elements are not in the document since before.

Finally, you need to be sure the entire ui is completely updated before doing other stuff, so you need to wait for the other sections to update too when clicking on and deleting items. And when you delete persona1, the default is selected automatically, delegates appears and getRightsRequest is fired. And then also for persona2, click on it will unmount again the delegates section, and the delete will then select automatically the default again, delegates section is mounted again and a new getRightsRequest is fired. We need to wait for all of this inside this test. 

```suggestion
		await user.click(screen.getByRole('button', { name: /add persona/i }));
		await waitFor(() =>
			expect(screen.getByRole('textbox', { name: /persona name/i })).toHaveDisplayValue(
				/new persona 1/i
			)
		);

		const persona1 = 'New Persona 1';
		const persona2 = 'New Persona 2';
		const persona3 = 'New Persona 3';
		expect(screen.getByText(persona1)).toBeVisible();

		await user.click(screen.getByRole('button', { name: /add persona/i }));
		await waitFor(() =>
			expect(screen.getByRole('textbox', { name: /persona name/i })).toHaveDisplayValue(persona2)
		);
		expect(screen.getByText(persona2)).toBeVisible();

		await user.click(screen.getByRole('button', { name: /add persona/i }));
		await waitFor(() =>
			expect(screen.getByRole('textbox', { name: /persona name/i })).toHaveDisplayValue(persona3)
		);
		expect(screen.getByText(persona3)).toBeVisible();

		await user.click(screen.getByText(persona1));
		await waitFor(() =>
			expect(screen.getByRole('textbox', { name: /persona name/i })).toHaveDisplayValue(persona1)
		);
		await user.click(screen.getByRole('button', { name: /delete/i }));
		let confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
		await user.click(confirmButton);
		await screen.findByText(/primary account settings/i);
		await screen.findByText('sendAs');
		expect(screen.queryByText(persona1)).not.toBeInTheDocument();

		await user.click(screen.getByText(persona2));
		await waitFor(() =>
			expect(screen.getByRole('textbox', { name: /persona name/i })).toHaveDisplayValue(persona2)
		);
		await user.click(screen.getByRole('button', { name: /delete/i }));
		confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
		await user.click(confirmButton);
		await screen.findByText(/primary account settings/i);
		await screen.findByText('sendAs');
		expect(screen.queryByText(persona2)).not.toBeInTheDocument();

		await user.click(screen.getByRole('button', { name: /save/i }));
```